IOPool/Input/src/PoolSource.h

```diff
@@ -80,6 +81,10 @@ namespace edm {
     bool bypassVersionCheck_;
     int const treeMaxVirtualSize_;
     RunNumber_t setRun_;
+    std::vector<RunNumber_t> setRunNumberForEachLumi_;
```

Comment:
I assume you mean for handling setRun_ and setRunNumberForEachLumi_.
I'll investigate this. My off the top of my head guess is that it could improve the code, but take considerable effort.